lib/numbers_in_words/to_number.rb

```diff
@@ -16,27 +16,39 @@ def language
     end
   end
 
-  def handle_negative text
-    -1 * (text.gsub(/^minus /, "")).in_numbers if text =~ /^minus /
+  def handle_negative(text, only_compress)
+    if text =~ /^minus/
+      stripped = text.gsub(/^minus/, "")
+      stripped_n = stripped.in_numbers(only_compress)
+      only_compress ? stripped_n.map{ |k| k * -1 } : -1 * stripped_n
+    end
   end
 
-  def in_numbers
-    text = to_s
+  def in_numbers(only_compress = false)
+    text = to_s.strip
+    return text.to_f if text =~ /^-?\d+(.\d+)?$/
 
     text = strip_punctuation text
-    i = handle_negative text
+
+    i = handle_negative(text, only_compress)
     return i if i
 
+    mixed = text.match /^(-?\d+(.\d+)?) (hundred|thousand|million|billion|trillion)$/
```

Comment:
These should come from the constant that defines them already (and is more comprehensive)